### PR TITLE
[BEAM-1694] Fix docstring inaccuracies in Python-SDK

### PR DESCRIPTION
**Pull request by tibkiss**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by tibkiss** [comment]
R: @aaltay  / @sb2nov

**Comment by sb2nov** [comment]
LGTM, thanks for fixing these 👍 

R: @aaltay for merging

**Comment by tibkiss** [comment]
Two things to note:
1) I changed `PartitionFn.partition_for` methods signature to match with the docstring. The idea is to be in sync with other classes.
2) CombinePerKey's Args seems invalid (but I'm not 100% sure): https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/core.py#L916-L921 .  If you also find it unnecessary I'll update the PR and remove those lines.

**Comment by tibkiss** [comment]
Thanks for the super-fast respone @sb2nov ! :)

**Comment by sb2nov** [comment]
My thoughts on the CombinePerKey:
We can leave the CombinePerKey as it is in this PR. I can see why it is unnecessary for a derived class which just uses the parent class constructor. But this is a style decision so we should get the opinion of others.

On that note, it might be actually good to validate that `fn` is an instance of type `CombineFn` here similar to a ParDo. https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/core.py#L596

**Comment by tibkiss** [comment]
@sb2nov : Thanks for your response. I think it makes sense to keep the doc about args in there: 
I initially thought  that it is a copy-paste mistake (due to lack of direct constructor), but your explanation & parent constructor explains it all. 

Your idea on validation is great: It'll remove the ambiguity caused by inherited constructor and also increase safety. Will create a Ticket / PR for that soon.

File: sdks/python/apache_beam/io/avroio.py [hunk 1]
```diff
@@ -129,6 +127,12 @@ def read_block_from_file(f, codec, schema, expected_sync_marker):
 
     Args:
       f: Avro file to read.
+      codec: Decompression codec to use.
```
Comment (aaltay) [inline sdks/python/apache_beam/io/avroio.py]: For consistency, I would use `The codec to use for block-level decompression.`

File: sdks/python/apache_beam/io/avroio.py [hunk 2]
```diff
@@ -129,6 +127,12 @@ def read_block_from_file(f, codec, schema, expected_sync_marker):
...
+        Supported codecs: 'null', 'deflate', 'snappy'
+      schema: Avro Schema definition represented as JSON string.
+      expected_sync_marker: Avro synchronization marker.
```
Comment (aaltay) [inline sdks/python/apache_beam/io/avroio.py]: Would this fit in two lines?

```
      expected_sync_marker: Avro synchronization marker. If the block's sync
        marker does not match with this parameter then ValueError is thrown.
```

Second question (to @sb2nov), should we start annotating code blocks inside comments for the them to show up nicely in pydocs? (like ValueError in this case.)

File: sdks/python/apache_beam/typehints/trivial_inference.py [hunk 3]
```diff
@@ -218,9 +218,10 @@ def infer_return_type(c, input_types, debug=False, depth=5):
   """Analyses a callable to deduce its return type.
 
   Args:
-    f: A Python function object to infer the return type of.
+    c: A Python callable to infer the return type of.
     input_types: A sequence of inputs corresponding to the input types.
     debug: Whether to print verbose debugging information.
+    depth: Maximal inspection depth of function calls during type inference
```
Comment (aaltay) [inline sdks/python/apache_beam/typehints/trivial_inference.py]: How about: `Maximum inspection depth during type inference.`

File: sdks/python/apache_beam/utils/retry.py [hunk 4]
```diff
@@ -143,7 +143,7 @@ def with_exponential_backoff(
       can be used so that the delays are not randomized.
     factor: The exponential factor to use on subsequent retries.
       Default is 2 (doubling).
-    max_delay_sec: Maximum delay (in seconds). After this limit is reached,
+    max_delay_secs: Maximum delay (in seconds). After this limit is reached,
       further tries use max_delay_sec instead of exponentially increasing
       the time. Defaults to 4 hours.
```
Comment (aaltay) [inline sdks/python/apache_beam/utils/retry.py]: Could you update this to `Defaults to 1 hour.`?

File: sdks/python/apache_beam/io/avroio.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/avroio.py]: +1 for annotating code blocks

Comment (tibkiss) [inline sdks/python/apache_beam/io/avroio.py]: I'll open a separate issue for annotating code blocks soon.

**Comment by tibkiss** [comment]
@aaltay : Thanks for your review. Updated the PR based on your comments.

**Comment by aaltay** [comment]
LGTM. Thank you.

**Pull request closed without merging**